src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFromFingerprintFacet.java

```diff
@@ -3,13 +3,30 @@
 import hudson.model.Fingerprint;
 
 /**
- * As a facet on a fingerprint that represents a docker image,
- * indicates that another image was produced by deriving from it. See "FROM" in Dockerfile.
- *  
- * @author Kohsuke Kawaguchi
+ * Facet representing the fact that one Docker image was derived from another.
+ * This corresponds to the {@code FROM} directive in a {@code Dockerfile}.
+ * As a facet on a fingerprint corresponding to the base image, refers to the derived image.
+ * As a facet on a fingerprint corresponding to the derived image, refers to the base image.
+ * Both fingerprints should be associated with the same build.
+ * @see DockerFingerprints#addFromFacet
  */
 public class DockerFromFingerprintFacet extends DockerRunPtrFingerprintFacet {
-    DockerFromFingerprintFacet(Fingerprint fingerprint, long timestamp, String imageId) {
+
+    private final String otherImageId;
+    private final boolean inverse;
+
+    DockerFromFingerprintFacet(Fingerprint fingerprint, long timestamp, String imageId, String otherImageId, boolean inverse) {
         super(fingerprint, timestamp, imageId);
+        this.otherImageId = otherImageId;
+        this.inverse = inverse;
+    }
+
+    public String getBaseImageId() {
+        return inverse ? otherImageId : getImageId();
```

Comment:
+1 to separate classes - but I think they should have a different method signature depending on if its pointing at the from/parent or the derived/child